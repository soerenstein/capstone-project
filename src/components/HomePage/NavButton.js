import React from 'react'
import styled from 'styled-components/macro'

export default function NavButton() {
  return (
    <div>
      <NavWrapper>
        <a href="/user-profile">
          <ButtonStyled>
            <IconStyled src={''} alt="" />
            <H2Styled>Profil</H2Styled>
          </ButtonStyled>
        </a>
        <a href="/create">
          <ButtonStyled>
            <IconStyled src={''} alt="" />
            <H2Styled>Kontakt hinzuf√ºgen</H2Styled>
          </ButtonStyled>
        </a>
        <a href="/list">
          <ButtonStyled>
            <IconStyled src={''} alt="" />
            <H2Styled>Kontaktliste</H2Styled>
          </ButtonStyled>
        </a>
        <a href="/favorites">
          <ButtonStyled>
            <IconStyled src={''} alt="" />
            <H2Styled>Favoriten</H2Styled>
          </ButtonStyled>
        </a>
      </NavWrapper>
    </div>
  )
}

const NavWrapper = styled.div`
  display: flex;
  width: 100%;
  flex-wrap: wrap;
  justify-content: space-evenly;
  align-items: center;
  margin-top: 20px;
`

const ButtonStyled = styled.button`
  width: 20vh;
  height: 20vh;
  background: white;
  margin: 10px;
  border-radius: 40px;
  box-shadow: 8px 6px 20px -8px white;
  border-style: none;
`

const IconStyled = styled.img`
  margin-top: 15px;
  vertical-align: top;
`

const H2Styled = styled.h2`
  font-size: 18px;
  vertical-align: bottom;
  margin-bottom: 0;
`
