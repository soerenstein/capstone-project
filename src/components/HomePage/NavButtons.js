import React from 'react'
import styled from 'styled-components/macro'
import { ReactComponent as UserProfile } from '../Design/img/userHomepage.svg'
import { ReactComponent as UserList } from '../Design/img/userListHomepage.svg'
import { ReactComponent as AddUser } from '../Design/img/userPlusHomepage.svg'
import { ReactComponent as FavoriteUser } from '../Design/img/FavoriteUserHomepage.svg'
import { NavLink } from 'react-router-dom'

export default function NavButtons() {
  return (
    <>
      <NavWrapper>
        <NavLink to="/user-profile">
          <NavButtonWrapper>
            <UserProfile />
            <H2Styled>Profil</H2Styled>
          </NavButtonWrapper>
        </NavLink>
        <NavLink to="/create">
          <NavButtonWrapper>
            <AddUser />
            <H2Styled>Kontakt hinzuf√ºgen</H2Styled>
          </NavButtonWrapper>
        </NavLink>
        <NavLink to="/list">
          <NavButtonWrapper>
            <UserList />
            <H2Styled>Kontaktliste</H2Styled>
          </NavButtonWrapper>
        </NavLink>
        <NavLink to="/favorites">
          <NavButtonWrapper>
            <FavoriteUser />
            <H2Styled>Favoriten</H2Styled>
          </NavButtonWrapper>
        </NavLink>
      </NavWrapper>
    </>
  )
}

const NavWrapper = styled.div`
  position: absolute;
  bottom: 50px;
  display: flex;

  width: 100%;
  flex-wrap: wrap;
  justify-content: space-evenly;
  margin-top: 20px;
  & a {
    text-decoration: none;
  }
  & svg {
    margin-top: 20%;
  }
`

const NavButtonWrapper = styled.div`
  width: 22vh;
  height: 22vh;
  display: grid;
  grid-template-rows: 60% 40%;
  justify-items: center;
  margin: 10px;
  align-content: center;
  text-align: center;
  background: transparent 0% 0% no-repeat padding-box;
  box-shadow: 0px 3px 10px #00000029;
  border-radius: 30px;
`

const H2Styled = styled.h2`
  font-size: 1.1em;
  vertical-align: bottom;
  color: var(--black);
  font-weight: 500;
  margin-top: 0;
  margin-bottom: 0;
`
