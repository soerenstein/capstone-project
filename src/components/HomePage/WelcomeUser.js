import React from 'react'
import styled from 'styled-components/macro'

export default function WelcomeUser({ user }) {
  const { firstName } = user

  return (
    <WelcomeWrapper>
      {firstName === undefined ? (
        <>
          <StyledWelcome>Moin!</StyledWelcome>
        </>
      ) : (
        <>
          <StyledWelcome>Moin</StyledWelcome>
          <StyledWelcome>{firstName}</StyledWelcome>
        </>
      )}
    </WelcomeWrapper>
  )
}

const WelcomeWrapper = styled.div`
  position: absolute;
  top: 100px;
  left: 40px;
`

const StyledWelcome = styled.h2`
  margin: 0px;
  color: white;
`
