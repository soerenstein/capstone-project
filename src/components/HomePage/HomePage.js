import React from 'react'
import WelcomeUser from './WelcomeUser'
import NavButtons from './NavButtons'
import TopLeftCard from '../Design/CardWrapper/TopLeftCard'

export default function HomePage({ user }) {
  return (
    <>
      <TopLeftCard />
      <WelcomeUser user={user} />
      <NavButtons />
    </>
  )
}
