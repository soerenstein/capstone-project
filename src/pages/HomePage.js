import React from 'react'
import WelcomeUser from '../components/HomePage/WelcomeUser'
import NavButtons from '../components/HomePage/NavButtons'
import TopLeftCard from '../components/Design/CardWrapper/TopLeftCard'

export default function HomePage({ user }) {
  return (
    <>
      <TopLeftCard />
      <WelcomeUser user={user} />
      <NavButtons />
    </>
  )
}
