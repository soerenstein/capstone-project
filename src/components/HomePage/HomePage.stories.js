import React from 'react'
import HomePage from './HomePage'

export default {
  title: 'HomePage',
  component: HomePage,
}

const HomeNav = (args) => <HomePage {...args} />

export const HomeNavigation = HomeNav.bind({})
HomeNavigation.args = {
  user: { firstName: 'Maxi' },
}
