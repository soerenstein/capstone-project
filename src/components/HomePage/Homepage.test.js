import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { MemoryRouter } from 'react-router-dom'
import HomePage from './HomePage'

describe('The homepage', () => {
  it('renders correctly', () => {
    const tree = render(
      <MemoryRouter>
        <HomePage />
      </MemoryRouter>
    )
    expect(tree).toMatchSnapshot()
  })
})

describe('The homepage', () => {
  it(' renders correctly', () => {
    const { getByText } = render(
      <MemoryRouter>
        <HomePage />
      </MemoryRouter>
    )
    expect(getByText('Moin.')).toBeInTheDocument()
    expect(getByText('Profil')).toBeInTheDocument()
    expect(getByText('Kontakt hinzufügen')).toBeInTheDocument()
    expect(getByText('Kontaktliste')).toBeInTheDocument()
    expect(getByText('Favoriten')).toBeInTheDocument()
  })
})

describe('The homepage', () => {
    const user = {
        firstName: 'Max',
      }
    it('renders with username', () => {
      const { getByText } = render(
        <MemoryRouter>
          <HomePage user={user} />
        </MemoryRouter>
      )
      expect(getByText('Moin')).toBeInTheDocument()
      expect(getByText('Max')).toBeInTheDocument()
      expect(getByText('Profil')).toBeInTheDocument()
      expect(getByText('Kontakt hinzufügen')).toBeInTheDocument()
      expect(getByText('Kontaktliste')).toBeInTheDocument()
      expect(getByText('Favoriten')).toBeInTheDocument()
    })
  })
  