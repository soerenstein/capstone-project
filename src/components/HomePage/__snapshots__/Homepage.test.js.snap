// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The homepage renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="TopLeftCard__StyledTopCard-sc-1tvhhfn-1 jUbhAn"
      >
        <svg
          class="TopLeftCard__StyledLogo-sc-1tvhhfn-0 gnoeTV"
        >
          logo.svg
        </svg>
      </div>
      <div
        class="WelcomeUser__WelcomeWrapper-sc-1uriybj-0 fmJcjV"
      >
        <h2
          class="WelcomeUser__StyledWelcome-sc-1uriybj-1 ljakRc"
        >
          Moin.
        </h2>
      </div>
      <div
        class="NavButtons__NavWrapper-be11z7-0 bkLDAC"
      >
        <a
          href="/user-profile"
        >
          <div
            class="NavButtons__NavButtonWrapper-be11z7-1 kvcfxI"
          >
            <svg>
              userHomepage.svg
            </svg>
            <h2
              class="NavButtons__H2Styled-be11z7-2 hjdpHL"
            >
              Profil
            </h2>
          </div>
        </a>
        <a
          href="/create"
        >
          <div
            class="NavButtons__NavButtonWrapper-be11z7-1 kvcfxI"
          >
            <svg>
              userPlusHomepage.svg
            </svg>
            <h2
              class="NavButtons__H2Styled-be11z7-2 hjdpHL"
            >
              Kontakt hinzufügen
            </h2>
          </div>
        </a>
        <a
          href="/list"
        >
          <div
            class="NavButtons__NavButtonWrapper-be11z7-1 kvcfxI"
          >
            <svg>
              userListHomepage.svg
            </svg>
            <h2
              class="NavButtons__H2Styled-be11z7-2 hjdpHL"
            >
              Kontaktliste
            </h2>
          </div>
        </a>
        <a
          href="/favorites"
        >
          <div
            class="NavButtons__NavButtonWrapper-be11z7-1 kvcfxI"
          >
            <svg>
              FavoriteUserHomepage.svg
            </svg>
            <h2
              class="NavButtons__H2Styled-be11z7-2 hjdpHL"
            >
              Favoriten
            </h2>
          </div>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="TopLeftCard__StyledTopCard-sc-1tvhhfn-1 jUbhAn"
    >
      <svg
        class="TopLeftCard__StyledLogo-sc-1tvhhfn-0 gnoeTV"
      >
        logo.svg
      </svg>
    </div>
    <div
      class="WelcomeUser__WelcomeWrapper-sc-1uriybj-0 fmJcjV"
    >
      <h2
        class="WelcomeUser__StyledWelcome-sc-1uriybj-1 ljakRc"
      >
        Moin.
      </h2>
    </div>
    <div
      class="NavButtons__NavWrapper-be11z7-0 bkLDAC"
    >
      <a
        href="/user-profile"
      >
        <div
          class="NavButtons__NavButtonWrapper-be11z7-1 kvcfxI"
        >
          <svg>
            userHomepage.svg
          </svg>
          <h2
            class="NavButtons__H2Styled-be11z7-2 hjdpHL"
          >
            Profil
          </h2>
        </div>
      </a>
      <a
        href="/create"
      >
        <div
          class="NavButtons__NavButtonWrapper-be11z7-1 kvcfxI"
        >
          <svg>
            userPlusHomepage.svg
          </svg>
          <h2
            class="NavButtons__H2Styled-be11z7-2 hjdpHL"
          >
            Kontakt hinzufügen
          </h2>
        </div>
      </a>
      <a
        href="/list"
      >
        <div
          class="NavButtons__NavButtonWrapper-be11z7-1 kvcfxI"
        >
          <svg>
            userListHomepage.svg
          </svg>
          <h2
            class="NavButtons__H2Styled-be11z7-2 hjdpHL"
          >
            Kontaktliste
          </h2>
        </div>
      </a>
      <a
        href="/favorites"
      >
        <div
          class="NavButtons__NavButtonWrapper-be11z7-1 kvcfxI"
        >
          <svg>
            FavoriteUserHomepage.svg
          </svg>
          <h2
            class="NavButtons__H2Styled-be11z7-2 hjdpHL"
          >
            Favoriten
          </h2>
        </div>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
